analyses/molecular-subtyping-MB/01-filter-and-batch-correction.R

```diff
@@ -30,73 +28,55 @@ data_dir <- file.path(root_dir, "data")
-stranded.file <- file.path(root_dir, "analyses", "collapse-rnaseq", "results","pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds")
-terms.file <- file.path("input", "mb_subtyping_path_dx_strings.json")
+clin_file <- file.path(data_dir, "histologies.tsv")
+exprs_file <- file.path(data_dir, "gene-expression-rsem-tpm-collapsed.rds")
+terms_file <- file.path("input", "mb_subtyping_path_dx_strings.json")
 
 # Read in the JSON file that contains the strings we'll use to include or
 # exclude samples for subtyping - see 00-mb-select-pathology-dx
-path_dx_list <- jsonlite::fromJSON(terms.file)
+path_dx_list <- jsonlite::fromJSON(terms_file)
 
-# output files
-uncorrected.file <- file.path(output_dir, paste0(output_prefix, ".rds"))
-corrected.file <- file.path(output_dir, paste0(output_prefix, "-batch-corrected.rds"))
+# output file
+uncorrected_file <- file.path(output_dir, paste0(output_prefix, ".rds"))
 
-# expression from polya and stranded data
-polya <- readRDS(polya.file)
-stranded <- readRDS(stranded.file)
+# collapsed expression matrix
+exprs_mat <- readRDS(exprs_file)
 
 # read and subset clinical file to MB samples
 clin <- read.delim(clin_file, stringsAsFactors = F)
 
 # Filter to medulloblastoma samples only based on criteria determined in 00-mb-select-pathology-dx
-clin.mb  <- clin %>%
+clin_mb  <- clin %>%
   # Inclusion on the basis of strings in pathology_diagnosis and
   # if 'Other', then based on pathology_free_text_diagnosis
   filter(pathology_diagnosis %in% path_dx_list$exact_path_dx |
-         (pathology_diagnosis == "Other" & str_detect(str_to_lower(pathology_free_text_diagnosis), paste0(path_dx_list$include_free_text, collapse = "|"))))
+           (pathology_diagnosis == "Other" & str_detect(str_to_lower(pathology_free_text_diagnosis), paste0(path_dx_list$include_free_text, collapse = "|"))))
 
 # Write to TSV for use later
-readr::write_tsv(clin.mb, file.path("input", "subset-mb-clinical.tsv"))
+readr::write_tsv(clin_mb, file.path("input", "subset-mb-clinical.tsv"))
 
 # Keep to only RNA-seq for these next step
-clin.mb.rnaseq <- clin.mb %>%
+clin_mb_rnaseq <- clin_mb %>%
   dplyr::filter(experimental_strategy == "RNA-Seq")
 
-# Make this a expression so we can match any of these
-biospecimen_ids <- paste0(clin.mb.rnaseq$Kids_First_Biospecimen_ID, collapse = "|")
-
-# Select only the mb biospecimens in polya
-polya.mb <- polya %>%
-  dplyr::select(dplyr::matches(biospecimen_ids)) %>%
-  rownames_to_column('gene')
-
-# Select only the mb biospecimens in stranded
-stranded.mb <-  stranded %>%
-  dplyr::select(dplyr::matches(biospecimen_ids)) %>%
-  rownames_to_column('gene')
-
-# inner_join by gene, turn into a matrix
-all.exprs.mb <- polya.mb %>%
-  inner_join(stranded.mb, by = "gene") %>%
-  column_to_rownames('gene') %>%
-  as.matrix()
+# Select only the mb biospecimens in expression mat
+exprs_mat_mb <- exprs_mat %>%
+  dplyr::select(dplyr::matches(clin_mb_rnaseq$Kids_First_Biospecimen_ID))
```

Comment:
When running the script, it gives me 
```Error in dplyr::matches(clin_mb_rnaseq$Kids_First_Biospecimen_ID) : 
  is_string(match) is not TRUE
Calls: %>% ... vars_select_eval -> map_if -> map -> .f -> <Anonymous> -> stopifnot
```
I think we can just do the following since we do expect all specimens to have expression data:
```suggestion
  dplyr::select(clin_mb_rnaseq$Kids_First_Biospecimen_ID)
```

Reply:
That is weird, this was in the original script as well and worked then and now as well. However, I will remove it as it does not change any results.